cdap-security/src/main/java/io/cdap/cdap/security/auth/UserIdentity.java

```diff
@@ -53,28 +59,55 @@ public static Schema getCurrentSchema() {
     }
   }
 
+  /**
+   * Identifies the type of {@link UserIdentity}. This is used to distinguish between access tokens generated
+   * for the external authentication server and the access tokens generated for use by internal services.
+   */
+  public enum IdentifierType {
+    /**
+     * EXTERNAL represents access tokens which may be issued to end-users.
+     * External access tokens are checked by the authorization extension.
+     */
+    EXTERNAL,
+    /**
+     * INTERNAL represents access tokens used by system services.
+     * Internal access tokens typically have higher-privileged access and are checked by the
+     * {@link io.cdap.cdap.security.authorization.InternalAccessEnforcer}.
+     */
+    INTERNAL
+  }
+
   private final String username;
+  private final IdentifierType identifierType;
   private final List<String> groups;
   private final long issueTimestamp;
   private final long expireTimestamp;
 
-  public UserIdentity(String username, Collection<String> groups, long issueTimestamp,
-                      long expireTimestamp) {
+  public UserIdentity(String username, IdentifierType identifierType, Collection<String> groups,
+                      long issueTimestamp, long expireTimestamp) {
     this.username = username;
+    this.identifierType = identifierType;
     this.groups = ImmutableList.copyOf(groups);
     this.issueTimestamp = issueTimestamp;
     this.expireTimestamp = expireTimestamp;
   }
 
   /**
-   * Returns the username for this identity.
+   * Returns the username for this access token.
```

Comment:
Why do we name class identity and refer to it as access token? It's confusing.

Reply:
I don't think it's a token until after it's signed. As an object it's still just identity, potentially unverified

Reply:
I did a refactor of the names, but then reverted it. These are remnants of that refactor, so I'll go ahead and revert them.